### ISSUE TITLE
make uninstall does not remove all ejabberd directories

### ISSUE THREAD

**Issue by nandlab** [processone/ejabberd#4133]
## Environment

- ejabberd version: commit [a57bdff](https://github.com/processone/ejabberd/commit/a57bdfffb715abd1334045bf7b33c113c1b46ca0)
- Erlang version: 13.1.5
- OS: Devuan GNU/Linux 5 (daedalus)

## Compilation
```
./autogen.sh
./configure
make
```

## Problem
`sudo make install` installs the ejabberd library files to /usr/local/lib/ejabberd-23.10.34.
`sudo make uninstall` tries to removes the directory /usr/local/lib/ejabberd which does not exist.
Therefore, /usr/local/lib/ejabberd-23.10.34 stays after uninstall. This is not expected.

Can you please install the library files to /usr/local/lib/ejabberd?

**Comment by badlop** [comment]
ejabberd and all the other libraries are installed with their version number, following the Erlang/OTP convention. The problem is that `make uninstall` does not follow that convention. This was reported long ago in https://github.com/processone/ejabberd/issues/1496

I tried to write a patch, but it still left some files there in the system... https://github.com/processone/ejabberd/issues/1496#issuecomment-282039442

Yes, it would be great if somebody fixes `make uninstall` to really uninstall the binary files.

Some workarounds:
```
./configure --prefix=/some/path
make
make install
```

Or using OTP releases, which includes erlang from your system:
```
./configure
make
make rel
mkdir -p /opt/ejabberd
tar -xzvf _build/prod/rel/ejabberd/ejabberd-23.10.64.tar.gz -C /opt/ejabberd
```

If you repeat this same process with a newer ejabberd version, it will install the binary files with the versioned directories, so they don't conflict, and the start script will use the newer binaries. Of course, you can delete `erts-14.2/`, `lib/` and `releases/` before installing a new version

**Comment by nandlab** [comment]
Thank you for the info!
What about the apt package repository? It installed files in /opt/ejabberd-23.10. What will happen when there is an upgrade to 23.11, will the /opt/ejabberd-23.10 directory stay or will it be automatically removed?

**Comment by badlop** [comment]
Let's see a summary of how you upgrade to a new ejabberd version:

1. When you install a new version, the binary files, any additional files, and the start script, will be available in a new location (`/opt/ejabberd-24.07` or whatever).

2. At that moment it is highly recommended that you make a backup of your ejabberd data files. I mean, the files in `/opt/ejabberd/`. This includes your configuration files, your mnesia database files, your certificates...

3. Then you should start using the new start script, which uses the new binaries etc.

4. The old directory is not automatically removed. If you suffer any problem running the new ejabberd version, you can recover your data files from step 2, and revert step 3. That way you can continue using the old ejabberd version until you can ask for help and solve the problems in the new ejabberd version.

**Comment by badlop** [comment]
By the way, as  `make uninstall` is buggy and I found no way to fix it, [ejabberd 24.02](https://www.process-one.net/blog/ejabberd-24-02/) introduced a new experimental alternative: `make uninstall-rel`
